### DISCUSSION TITLE
I am curious, where does the info to Flymake is coming from?

### DISCUSSION THREAD

**Discussion by refaelsh** [joaotavora/eglot#1244] [Q&A] [answered]
1. I am assuming it's from Eglot, e.g., Eglot acts as a backend for Flymake, right?
2. Where does Eglot receive it from? Does it receive it from the LSP or some other external linting tool (for Haskell it would be `hlint` for example)?

Thanks in advance.

**Comment by joaotavora** [comment, marked as answer]
> I am assuming it's from Eglot, e.g., Eglot acts as a backend for Flymake, right?

Yes.

> Does it receive it from the LSP

Yes.  LSP server might get it from its own analysis or from tools like `hlint` or whatever.  I'm not familiar with the Haskell server at all.

**Comment by refaelsh** [comment]
I understand. Thank you very much for the answers @joaotavora.
